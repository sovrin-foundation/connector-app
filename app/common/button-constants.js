// @flow
export const DENY = 'Deny'
export const CONNECT = 'Connect'
