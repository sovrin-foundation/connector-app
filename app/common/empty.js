// @flow
// sometimes we need an empty object to use as a placeholder in destructuring
export default {}
