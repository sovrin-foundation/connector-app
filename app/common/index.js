// @flow
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
