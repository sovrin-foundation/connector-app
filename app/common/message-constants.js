// @flow
// for overall app message string, it can be error, success, warning etc.

export const TOUCH_ID_MESSAGE = 'Please confirm with Biometrics'
export const TOUCH_ID_NOT_AVAILABLE =
  'Biometrics is not supported on this device'

export const DEVICE_ENROLLMENT_ERROR = [
  'Enrollment Error',
  'Identifier or Seed or Push-notification token not present!',
]

export const PUSH_NOTIFICATION_PERMISSION_ERROR = [
  'Permission Error',
  'Push Notification permission needed! Please enable it from Settings.',
]

export const DUPLICATE_CONNECTION_ERROR = [
  'Attention!',
  'Duplicate Connection invitation!',
]

export const UNLOCKING_APP_WAIT_MESSAGE = 'Loading please wait...'

export const ENTER_PASS_CODE_MESSAGE = 'Enter passcode'

export const ENTER_YOUR_PASS_CODE_MESSAGE = 'Enter your passcode'

export const FIRST_BACKUP_TITLE = 'Set up data recovery now!'

export const FIRST_BACKUP_SUBTEXT = '2 easy steps, takes 3 minutes.'

export const SUBSEQUENT_BACKUP_TITLE =
  'You have new data that is not backed up.'

export const SUBSEQUENT_BACKUP_SUBTEXT =
  "Create another backup so you don't lose anything."

export const PASSPHRASE_GENERATION_ERROR =
  'Uh oh... Looks like something went wrong.'
