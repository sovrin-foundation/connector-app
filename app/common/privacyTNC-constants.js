// @flow

import { Platform } from 'react-native'

const isAndroid = Platform.OS === 'android'
export { EULA_URL as TermsAndConditionUrl } from '../eula/type-eula'
export const TermsAndConditionsTitle = 'Terms and Conditions'

export const PrivacyPolicyUrl = 'https://www.connect.me/privacy.html'
export const localPrivacyPolicySource = isAndroid
  ? 'file:///android_asset/external/connectme/privacy.html'
  : './privacy.html'
export const PrivacyPolicyTitle = 'Privacy Policy'
