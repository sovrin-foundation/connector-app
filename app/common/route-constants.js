// @flow
export const splashScreenRoute = 'SplashScreenHide'
export const homeRoute = 'Home'
export const authenticationRoute = 'Authentication'
export const connectionRoute = 'Connection'
export const settingsRoute = 'Settings'
export const settingsScreen = 'SettingsStack'
export const expiredTokenRoute = 'ExpiredToken'
export const qrCodeScannerRoute = 'QRCodeScanner'
export const lockPinSetupRoute = 'PinSetup'
export const lockTouchIdSetupRoute = 'TouchIdSetup'
export const lockPinSetupHomeRoute = 'PinSetupHome'
export const lockSelectionRoute = 'LockSelection'
export const lockEnterPinRoute = 'LockEnterPin'
export const lockSetupSuccessRoute = 'LockSetupSuccess'
export const lockEnterFingerprintRoute = 'LockEnterFingerprint'
export const claimOfferRoute = 'ClaimOffer'
export const homeTabRoute = 'TabHome'
export const connectionsTabRoute = 'TabHome'
export const credentialsTabRoute = 'TabCredentials'
export const discoverTabRoute = 'TabDiscover'
export const menuTabRoute = 'TabMenu'
export const settingsTabRoute = 'TabSettings'
export const qrCodeScannerTabRoute = 'TabQRCodeScanner'
export const proofRequestRoute = 'ProofRequest'
export const invitationRoute = 'InvitationRoute'
export const connectionHistoryRoute = 'ConnectionHistoryRoute'
export const connectionHistoryDetailsRoute = 'ConnectionHistoryDetailsRoute'
export const switchEnvironmentRoute = 'SwitchEnvironment'
export const lockAuthorizationRoute = 'LockAuthorizationRoute'
export const lockAuthorizationHomeRoute = 'LockAuthorizationHomeRoute'
export const waitForInvitationRoute = 'WaitForInvitationRoute'
export const connectionHistoryPendingRoute = 'ConnectionHistoryPendingRoute'
export const walletRoute = 'TokenWallet'
export const walletTabSendDetailsRoute = 'WalletTabSendDetails'
export const receiveTabRoute = 'WalletReceiveTab'
export const sendTabRoute = 'WalletSendTab'
export const historyTabRoute = 'WalletHistoryTab'
export const aboutAppRoute = 'AboutApp'
export const onfidoRoute = 'Onfido'
export const privacyTNCRoute = 'PrivacyTNC'
export const eulaRoute = 'EulaScreen'
export const genRecoveryPhraseRoute = 'GenerateRecoveryPhrase'
export const verifyRecoveryPhraseRoute = 'VerifyRecoveryPhrase'
export const exportBackupFileRoute = 'ExportBackupFile'
export const backupCompleteRoute = 'BackupComplete'
export const restoreRoute = 'RestoreStartScreen'
export const restoreWaitRoute = 'RestoreWaitScreen'
export const restorePassphraseRoute = 'RestorePassphrase'
export const backupErrorRoute = 'BackupErrorScreen'
