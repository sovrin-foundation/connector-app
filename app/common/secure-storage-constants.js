// @flow

export const PUSH_COM_METHOD = 'pushComMethod'
export const CONNECT_ME_STORAGE_KEY = '@connectmeDefaults'
export const IS_ALREADY_INSTALLED = `${CONNECT_ME_STORAGE_KEY}:IS_ALREADY_INSTALLED`
export const CONNECTIONS = 'CONNECTIONS'
export const CLAIM_MAP = `CLAIM_MAP`
export const WALLET_ENCRYPTION_KEY = 'walletEncryptionKey'
export const WALLET_BALANCE = 'walletBalance'
export const WALLET_ADDRESSES = 'walletAddresses'
export const WALLET_HISTORY = 'walletHistory'
export const LAST_SUCCESSFUL_BACKUP = 'lastSuccessfulBackup'
export const PASSPHRASE_SALT_STORAGE_KEY = 'passphraseSalt'
export const PASSPHRASE_STORAGE_KEY = 'passphrase'
