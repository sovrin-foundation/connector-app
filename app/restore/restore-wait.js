// @flow

import React, { PureComponent } from 'react'
import { Image, StyleSheet, Dimensions } from 'react-native'
import { Container, CustomView, CustomText, Icon } from '../components'
import { darkGray } from '../common/styles/constant'
import { connect } from 'react-redux'
import { restoreRoute, restoreWaitRoute, lockEnterPinRoute } from '../common'
import type { RestoreWaitScreenProps } from './type-restore'
import type { Store } from '../store/type-store'
import { RestoreStatus } from './type-restore'

export class RestoreWaitScreen extends PureComponent<
  RestoreWaitScreenProps,
  void
> {
  componentDidUpdate(prevProps: RestoreWaitScreenProps) {
    if (
      !this.props.restore.error &&
      this.props.restore.status === RestoreStatus.RESTORE_DATA_STORE_SUCCESS &&
      this.props.route === restoreWaitRoute
    ) {
      // TODO: the params have to be removed when the lockEnterPinRoute design is changed in according with the recovery screen.
      this.props.navigation.navigate(lockEnterPinRoute, {
        fromScreen: 'recovery',
      })
    }

    if (
      this.props.restore.error !== prevProps.restore.error &&
      this.props.route === restoreWaitRoute
    ) {
      //the navigation stack here is like Restore Start-> Restore Wait->Restore Start
      // So before going to Restore Start goBack two times and remove the routes from the stack
      this.props.navigation.navigate(restoreRoute)
    }
  }

  render() {
    return (
      <Container fifth safeArea center>
        <CustomView style={[styles.mainContainer]}>
          <Image
            source={require('../images/bkgCurve.png')}
            style={[styles.backgroundImage]}
          />

          <CustomView center>
            <CustomView style={[styles.strip]} />

            <Icon
              xxLarge
              style={[styles.iconStyle]}
              src={require('../images/dataRestore.png')}
            />
          </CustomView>

          <CustomView center>
            <CustomText
              bold
              center
              transparentBg
              style={[styles.textContainer]}
              heavy
            >
              Please wait while your data is restored.
            </CustomText>
          </CustomView>
        </CustomView>
      </Container>
    )
  }
}

const styles = StyleSheet.create({
  mainContainer: {
    marginTop: '-20%',
  },
  textContainer: {
    marginBottom: '6%',
    marginTop: '15%',
    paddingHorizontal: '5%',
    color: darkGray,
    fontSize: 23,
  },
  backgroundImage: {
    position: 'absolute',
    marginTop: '-58%',
    width: '100%',
    zIndex: -1,
  },

  iconStyle: {
    borderRadius: 80,
    padding: 15,
    backgroundColor: darkGray,
  },
  strip: {
    position: 'absolute',
    width: '110%',
    height: 8,
    backgroundColor: darkGray,
  },
})

const mapStateToProps = (state: Store) => {
  return {
    restore: state.restore,
    route: state.route.currentScreen,
  }
}

export default connect(mapStateToProps, null)(RestoreWaitScreen)
