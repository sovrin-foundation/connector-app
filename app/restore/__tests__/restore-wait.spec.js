// @flow
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { RestoreWaitScreen } from '../restore-wait'
import { getNavigation, getStore } from '../../../__mocks__/static-data'
import { restoreWaitRoute } from '../../common'

describe('RestoreWait screen', () => {
  function getProps() {
    return {
      navigation: getNavigation(),
      route: restoreWaitRoute,
      restore: getStore().getState().restore,
    }
  }

  function setup() {
    const props = getProps()
    return { props }
  }

  it('should render properly and match the snapshot', () => {
    const { props } = setup()
    const tree = renderer.create(<RestoreWaitScreen {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  //TODO add snapshot for error screen
})
