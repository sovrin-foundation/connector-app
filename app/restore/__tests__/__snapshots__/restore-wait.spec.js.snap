// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestoreWait screen should render properly and match the snapshot 1`] = `
<RCTSafeAreaView
  center={true}
  fifth={true}
  safeArea={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": "-20%",
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "marginTop": "-58%",
            "position": "absolute",
            "width": "100%",
            "zIndex": -1,
          },
        ]
      }
    />
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#505050",
            "height": 8,
            "position": "absolute",
            "width": "110%",
          }
        }
      />
      <View
        clearBg={true}
        src={1}
        style={
          Object {
            "backgroundColor": "#505050",
            "borderRadius": 80,
            "padding": 15,
          }
        }
        xxLarge={true}
      >
        <View
          style={Object {}}
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Array [
                Object {
                  "height": 90,
                  "width": 90,
                },
                null,
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "#505050",
              "fontSize": 23,
              "marginBottom": "6%",
              "marginTop": "15%",
              "paddingHorizontal": "5%",
            },
          ]
        }
      >
        Please wait while your data is restored.
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
